create_bot_knowledge.py

```diff
@@ -21,24 +21,277 @@
 import os
 import json
 
-from typing import List
-from typing import Tuple
+from typing import List, Tuple, Dict, Optional, Union, Set
 from pathlib import Path
 
-from github import Github
-from ogr.services.github import GithubService
-from ogr.abstract import Issue, IssueComment, IssueStatus, PullRequest, PRComment, PRStatus, CommitComment
+from github import Github, GithubObject, Issue, IssueComment, PullRequest, PullRequestReview
+from github.Repository import Repository
 
 _LOGGER = logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
 _GITHUB_ACCESS_TOKEN = os.getenv("GITHUB_ACCESS_TOKEN")
 
-PROJECTS = [
-    # AiCoE Team
-    # ("log-anomaly-detector", "aicoe"),
 
-    # # Thoth Team
+def connect_to_source(project: Tuple[str, str]) -> Repository:
+    """Connect to GitHub.
+
+    :param project: Tuple source repo and repo name.
+    """
+    # TODO: It should use only one library for source.
+
+    # Connect using PyGitHub
+    g = Github(_GITHUB_ACCESS_TOKEN)
+    repo_name = project[1] + "/" + project[0]
+    repo = g.get_repo(repo_name)
+
+    return repo
+
+
+def check_directory(knowledge_dir: Path):
+    """Check if directory for bot knowledge exists. If not, create one."""
+    if not knowledge_dir.exists():
+        _LOGGER.info(
+            "No knowledge from any repo has ever been created, creating new directory at %s" % knowledge_dir)
+        os.makedirs(knowledge_dir)
+
+
+issue_keywords = {'close',
```

Comment:
This variable and other below can be named in capital letters and moved to the beginning of the file.

Reply:
isn't it better to have them at the place we actually use them? I know that the constants tend to be at the beginning, but this seems like a more readable option to me.

Reply:
We have standards for a reason :)

Reply:
yeah i meant that indeed it should be named as constant (capitals + underscores)
but should it be at the beginning of a file? there seems to be no standard on that, or is it?
https://www.python.org/dev/peps/pep-0008/#constants

Reply:
You might need to reuse them in another file, it's easier to have them all at the beginning.

Reply:
It's a convention (I don't recall any standard on this). BTW, if you do not plan to adjust the set, you can declare it as a frozenset, as you use typing, you can declare it as final - something like (I haven't tried to run the code):

```python
from typing import Final
from typing import FrozenSet

ISSUE_KEYWORDS: Final[FrozenSet[str]] = frozenset({
                  'close',
                  'closes',
                  'closed',
                  'fix',
                  'fixes',
                  'fixed',
                  'resolve',
                  'resolves',
                  'resolved'
})